### PR TITLE
 GET for Driver Availability Controller (Driver Availability) 

### LINKED ISSUE #11: FEATURE: GET for Driver Availability Controller (Driver Availability)
Acceptance Criteria:

- [ ] There is an endpoint GET /api/driverAvailability that gets all availability submissions for the current user.

- [ ] GET /api/driverAvailability?id=123 that gets an availability submission if owned by the current user.

### PR DESCRIPTION
**Pull request by notaryaman**
adds GET endpoint for Driver Availability Controller, enter ID for in ID space to delete availability

working deployment: https://project-notaryaman-dev.dokku-16.cs.ucsb.edu/

image

image

closes #11

### CODE REVIEW

**Review by andrewpengucsb — CHANGES_REQUESTED** [review]
Same here, please update description

File: src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    @PreAuthorize("hasRole('ROLE_DRIVER')")
+    @GetMapping("/all")
+    public Iterable<DriverAvailability> allmenuitems() {
+        Iterable<DriverAvailability> drivall = driverAvailabilityRepository.findAll();
+        return drivall;
+    }
+
+
+    @Operation(summary= "Get Driver Availability")
+    @PreAuthorize("hasRole('ROLE_DRIVER')")
```
Comment (andrewpengucsb) [inline src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java:115]: Add admins 
Admins can get any
Drivers can only get their own

**Review by andrewpengucsb — CHANGES_REQUESTED** [review]

**Comment by andrewpengucsb** [comment]
Seems like this PR now has the same changes as #31

**Comment by notaryaman** [comment]
@andrewpengucsb we will still get points for it though right?

**Comment by andrewpengucsb** [comment]
Closing because became duplicate of #31

**Pull request closed without merging**